compiler/src/iree/compiler/DispatchCreation/test/fuse_encoding_ops_into_dispatch_regions.mlir

```diff
@@ -133,12 +143,16 @@ module {
   }
 }
 
+// CHECK-DAG:   #[[MAP1:.+]] = affine_map<(d0, d1, d2, d3) -> (d0, d3, d2)>
+// CHECK-DAG:   #[[MAP2:.+]] = affine_map<(d0, d1, d2, d3) -> (d0, d1, d3)>
+// CHECK-DAG:   #[[MAP3:.+]] = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2)>
+// CHECK-DAG:   #[[$ENCODING:.+]] = #iree_encoding.encoding<operand_index = 1 : index, op_type =  matmul, element_types = [f32, f32, f32], user_indexing_maps = [#[[MAP1]], #[[MAP2]], #[[MAP3]]], round_dims_to = array<i64: 32, 32, 32>>
 // CHECK-LABEL: @fusion_dynamic
 // CHECK-SAME:    {{.+}}: tensor<?x?x?xf32>, %[[D0:.+]]: index, %[[D1:.+]]: index, %[[D2:.+]]: index)
-// CHECK:       %[[DISPATCH0:.+]] = flow.dispatch.region -> (tensor<?x?x?xf32, #iree_encoding.encoding
+// CHECK:       %[[DISPATCH0:.+]] = flow.dispatch.region -> (tensor<?x?x?xf32, #[[$ENCODING]]>
 // CHECK-SAME:      {%[[D0]], %[[D1]], %[[D2]]}
 // CHECK:         %[[ADD:.+]] = linalg.generic
 // CHECK:         %[[SET_ENCODING:.+]] = iree_encoding.set_encoding
 // CHECK:         flow.return %[[SET_ENCODING]] :
 // CHECK:       }
-// CHECK:       util.return %[[DISPATCH0]] : tensor<?x?x?xf32, #iree_encoding.encoding
+// CHECK:       util.return %[[DISPATCH0]] : tensor<?x?x?xf32, #[[$ENCODING]]>
```

Comment:
What does `[[$ENCODING]]` mean?  Asking about the dollar sign here.

Reply:
If you capture the variables with `$` (i.e., see above `#[[$ENCODING:.+]]`), they are considered global and remain set throughout the file. Then you can get rid of the `CHECK-LABEL` issue that you're referring.

For more details, see https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-FileCheck-enable-var-scope

Reply:
nice! TIL.

Reply:
Yeah thats a footgun. It carries over `// -----` . So you could have

```
CHECK: #[[$<encoding-1-name>.+]] = 

// -----

CHECK: #[[$<encoding-2-name>:.+]] = 
 // [[$ENCODING1]] is still visible here.

```

So lets say you happened to use the same name for `<encoding-1-name>` and `<encoding-2-name>` then if you miss matching `encoding-2` FileCheck will think you meant to use `encoding-1`. Its really tricky to triage these.